compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/GenerateClosuresForDefaultValues.java

```diff
@@ -1526,14 +1569,16 @@ private <E extends BLangStatement> List<E> rewriteStmt(List<E> nodeList, SymbolE
         Queue<BLangSimpleVariableDef> previousQueue = this.queue;
         this.queue = new LinkedList<>();
         int size = nodeList.size();
+        rewrite(typeNodeOfRecordField, env);
```

Comment:
Why do we need to add `typeNodeOfRecordField` as an instance variable and visit it like this instead of doing it in-line?

Reply:
?

Reply:
```ballerina
function testScopingRules() {

    record {
        int a;
        record {
            int p = 10;
        } c = {};
    } f2 = {a: 20};

}
```
For the above scenario, the closure for P should be generated within the closure of C.

```ballerina
function desugaredTestScopingRules() {
    function () returns  record {int p = 10;} c = function() returns  record {int p = 10;} {
        function () returns int p = function () returns int {
            return 10;
        };
        return {p: p()};
    };

    record {
        int a;
        record {
            int p = 10;
        } c = {};
    } f2 = {a: 20};
}

```
To achieve this, we need to generate the closure for P while visiting the closure of C. To implement this logic, I have visit  the type node of record literal separately.

Reply:
No, right? How do we call p's closure only, if required?

Reply:
The following crashes at the moment. 

```ballerina
public function main() {
    record {
        record {
            int c = 10;
        } b = {};
    } _ = {b: {}};
}
```

Reply:
Fixed. Added tests also